backend/src/services/podcast/podcast.service.ts

```diff
@@ -186,6 +194,36 @@ class Podcast {
 
     return this.#podcastRepository.getAllByUserId(filterParams);
   }
+
+  public async delete(id: number): Promise<TPodcast> {
+    const podcast = await this.#podcastRepository.getById(id);
+
+    if (!podcast) {
+      throw new HttpError({
+        status: HttpCode.NOT_FOUND,
+        message: ErrorMessage.PODCAST_NOT_FOUND,
+      });
+    }
+
+    await this.#podcastRepository.delete(id);
+
+    const episodes = await this.#episodeService.getAllByPodcastId(id);
+    const isEpisodesExist = Boolean(episodes);
+
+    if (isEpisodesExist) {
+      await this.#episodeService.deleteAllByPodcastId(id);
+    }
```

Comment:
```suggestion
      await this.#episodeService.deleteAllByPodcastId(id);
```
We don't need this checks, if there are no episodes episodeService.deleteAllByPodcastId(id) will do nothing.

Reply:
I think this condition will not allow additional access to the database